### PR TITLE
Trace constructors

### PR DESCRIPTION
**Pull request by fitzgen**
r? @emilio

### CODE REVIEW

File: src/ir/comp.rs [hunk 1]
```diff
@@ -978,5 +978,9 @@ impl TypeCollector for CompInfo {
         for method in self.methods() {
             types.insert(method.signature);
         }
+
+        for &ctor in self.constructors() {
```
Comment (emilio) [inline src/ir/comp.rs]: Heh, so we did track them separately :)

**Comment by fitzgen** [comment]
Hrm.... looks like we have a lot of unresolved type refs... Not sure why this didn't trigger locally :-/

```
---- header_typeref_hpp stdout ----

	thread 'header_typeref_hpp' panicked at 'internal error: entered unreachable code', src/ir/ty.rs:925

---- header_union_dtor_hpp stdout ----

	thread 'header_union_dtor_hpp' panicked at 'internal error: entered unreachable code', src/ir/ty.rs:925

---- header_union_template_hpp stdout ----

	thread 'header_union_template_hpp' panicked at 'internal error: entered unreachable code', src/ir/ty.rs:925

---- header_template_hpp stdout ----

	thread 'header_template_hpp' panicked at 'internal error: entered unreachable code', src/ir/ty.rs:925

---- header_vector_hpp stdout ----

	thread 'header_vector_hpp' panicked at 'internal error: entered unreachable code', src/ir/ty.rs:925

---- header_union_with_big_member_h stdout ----

	thread 'header_union_with_big_member_h' panicked at 'internal error: entered unreachable code', src/ir/ty.rs:925

---- header_using_hpp stdout ----

	thread 'header_using_hpp' panicked at 'internal error: entered unreachable code', src/ir/ty.rs:925

---- header_variadic_method_hpp stdout ----

	thread 'header_variadic_method_hpp' panicked at 'internal error: entered unreachable code', src/ir/ty.rs:925

---- header_whitelist_basic_hpp stdout ----

	thread 'header_whitelist_basic_hpp' panicked at 'internal error: entered unreachable code', src/ir/ty.rs:925

---- header_use_core_h stdout ----

	thread 'header_use_core_h' panicked at 'internal error: entered unreachable code', src/ir/ty.rs:925

---- header_what_is_going_on_hpp stdout ----

	thread 'header_what_is_going_on_hpp' panicked at 'internal error: entered unreachable code', src/ir/ty.rs:925

---- header_whitelist_fix_hpp stdout ----

	thread 'header_whitelist_fix_hpp' panicked at 'internal error: entered unreachable code', src/ir/ty.rs:925

---- header_vtable_recursive_sig_hpp stdout ----

	thread 'header_vtable_recursive_sig_hpp' panicked at 'internal error: entered unreachable code', src/ir/ty.rs:925
```

**Comment by fitzgen** [comment]
Trying without that commit.

@bors-servo r=emilio

**Comment by emilio** [comment]
Hmm... That seems pretty much wrong to me, can you file an issue on it?

**Comment by emilio** [comment]
I can't reproduce either fwiw, may be a 3.8 thing? (wat)

**Comment by fitzgen** [comment]
Filed #450

**Pull request merged**